Remove reference of kfdef

zdtsw opened the pull request:
## Description
cleanup reference of kfdef when dashboard has support

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Merge criteria:
<!--- This PR will be merged by any repository approver when it meets all the points in the checklist -->
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

VaishnaviHire commented:
> > Does this PR need a rebase ? I see changes that are not related to removal of KFdef
> 
> it is based on latest commit, do we still need "`updatefromLegacyVersion()`" i do not see it is used and still it refers to `kfdef` so i commented it out and cleaned its package.
> 
> otherwise, it is only "." in the description i added, which are not directly for kfdef. all the rest is the `kfdef` cleanup.

You can remove the function, we can add it later when we move to stable version.

zdtsw commented:
> > > Does this PR need a rebase ? I see changes that are not related to removal of KFdef
> > 
> > 
> > it is based on latest commit, do we still need "`updatefromLegacyVersion()`" i do not see it is used and still it refers to `kfdef` so i commented it out and cleaned its package.
> > otherwise, it is only "." in the description i added, which are not directly for kfdef. all the rest is the `kfdef` cleanup.
> 
> You can remove the function, we can add it later when we move to stable version.

deleted.

VaishnaviHire reviewed (approved):
/lgtm
